svm/src/account_loader.rs

```diff
@@ -97,23 +100,39 @@ pub struct FeesOnlyTransaction {
 
 #[cfg_attr(feature = "dev-context-only-utils", derive(Clone))]
 pub(crate) struct AccountLoader<'a, CB: TransactionProcessingCallback> {
-    account_overrides: Option<&'a AccountOverrides>,
     pub(crate) program_cache: ProgramCacheForTxBatch,
     program_accounts: HashMap<Pubkey, (&'a Pubkey, u64)>,
+    account_cache: AHashMap<Pubkey, AccountCacheItem>,
     callbacks: &'a CB,
     pub(crate) feature_set: Arc<FeatureSet>,
 }
 impl<'a, CB: TransactionProcessingCallback> AccountLoader<'a, CB> {
-    pub fn new(
+    pub fn new_with_account_cache_capacity(
         account_overrides: Option<&'a AccountOverrides>,
         program_cache: ProgramCacheForTxBatch,
         program_accounts: HashMap<Pubkey, (&'a Pubkey, u64)>,
         callbacks: &'a CB,
         feature_set: Arc<FeatureSet>,
+        capacity: usize,
     ) -> AccountLoader<'a, CB> {
+        let mut account_cache = AHashMap::with_capacity(capacity);
+
+        if let Some(slot_history) =
+            account_overrides.and_then(|overrides| overrides.get(&slot_history::id()))
+        {
+            // We set `inspected_as_writable` to `true` because this never needs to be inspected.
+            account_cache.insert(
+                slot_history::id(),
+                AccountCacheItem {
+                    account: slot_history.clone(),
+                    inspected_as_writable: true,
+                },
+            );
+        }
```

Comment:
after this pr lands, we can contemplate a more restricted interface for passing only this account through. maybe `slot_history: Option<AccountSharedData>` on the transaction processing environment

bit of an aside, but the `sol_get_sysvar` syscall we added in 2.0 does not support `SlotHistory` by design, we only needed `StakeHistory` and `SlotHashes`. so the good news is it doesnt introduce a mechanism to bypass this override. the possibly better news is... if neither me nor joe needed to add `SlotHistory`, does anything actually use it? i have no idea. maybe it can be deprecated